asn1crypto/x509.py

```diff
@@ -947,8 +947,10 @@ def native(self):
                 for type_val in rdn:
                     field_name = type_val['type']
                     if field_name in self._native:
-                        self._native[field_name] = [self._native[field_name]]
-                        self._native[field_name].append(type_val['value'])
+                        existing = self._native[field_name]
+                        if not isinstance(existing, list):
+                            existing = self._native[field_name] = [existing]
+                        existing.append(type_val['value'])
                     else:
```

Comment:
I doubt if type['val'] may be list because as i understand it should be one of asn.1 simple types. However if we receive list as value it's questionable that we should flatten them and in this case it's better to track seen fields explicitly. What do you think?